projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx

```diff
@@ -182,15 +183,52 @@ export default class DomEventHandler extends Component {
 		}
 	};
 
-	scrollOverlayToTop = event => {
+	fixBodyScroll = event => {
 		// NOTE: the propertyName need to be aligned with the animation
 		if ( event?.propertyName !== 'opacity' ) {
 			return;
 		}
-		// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
-		window?.scrollTo( 0, 0 );
+
+		if ( this.props.isVisible ) {
+			this.preventBodyScroll();
+			// This is to fix search input on the overlay is not visible on mobile devices.
+			// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
+			window?.scrollTo( 0, 0 );
+		} else {
+			this.restoreBodyScroll();
+		}
 	};
 
+	preventBodyScroll() {
+		this.top = parseInt( window.scrollY ) || 0;
+		// Keep body at the same position when overlay is open.
+		document.body.style.top = `-${ this.top - this.getWpAdminBarComputedHeight() }px`;
+		// Make body in the center.
+		document.body.style.left = 0;
+		document.body.style.right = 0;
+		// Make body not scrollable.
+		document.body.style.position = 'fixed';
+	}
+
+	restoreBodyScroll() {
+		// Restore body scroll.
+		document.body.style.top = '';
+		document.body.style.left = '';
+		document.body.style.right = '';
+		document.body.style.position = '';
+		// Restore body position.
+		window.scrollTo( 0, this.top );
+	}
+
+	getWpAdminBarComputedHeight() {
+		const $wpAdminBar = document.getElementById( WP_ADMIN_BAR_ID );
```

Comment:
Should we have the `$` at the start of this var name?

Reply:
Old habit of add $ to a variable holding a node, will change this.

Reply:
The code is not necessary anymore.